### ISSUE TITLE
IPFS DDC - Exchange of ideas / collaborate

### ISSUE THREAD

**Issue by satazor** [IBM/ipfs-social-proof#49]
I've came across this project and we, at the [IPFS DDC](https://github.com/ipfs/dynamic-data-and-capabilities/) working group, will be doing something very similar. To give you some context, the IPFS DDC works towards offering libraries for developers to build DApps. Having an Identity solution is something we have been researching in the past few months. We have actually made a RFC for people to read and give feedback: https://github.com/ipfs-shipyard/peer-star/pull/15/files. The RFC culminates with the proposal of creating the IDM (Identity Manager):

- Allows to create/import identities
- Allows managing identities, including attaching social proofs to them
- Allows typical login workflows for users to login to DApps by choosing one identity
- Allow signing workflows between the DApps and the logged in identity
- Embraces open standards, such as DIDs, DID-Auth and Verifiable Claims

I would kindly ask if you could read the RFC and comment on the project similarities/differences.
Also, have you looked into https://github.com/jonnycrunch/ipid?  It's a DID method implemented on top of IPFS.

There's a lot of similarities between this project and IDM. I wonder if we could exchange ideas and perhaps even share code. Looking forward to see where this project is going and how we could collaborate!

**Comment by daviddahl** [comment]
I have been watching many IPFS repos in order to try and understand the scope and previous work of your Identity efforts. It has been extremely educational. I am very interested in collaboration and will be reading up on all of these repos and projects / working groups I may have missed in my perusal of IPFS repos. Thank you for pointing all of this out.

I do want to support whatever identity standards that are deemed important. I have been reading up on W3C's VC and ERC725, etc. I do believe all of the identity functionality can be done in IPFS w/ proofs a'la Keybase. However, the eventual UX is key. Some folks don't believe the ID proofs concept is "easy enough" for the typical internet user.  

I have been looking at this project as a way to quickly establish the "`boring`" Identity (and possibly user discovery) parts of a Dapp - without much overhead. At least the ID API. Peer discovery is also important and I was planning on building off of this a hacker news/reddit-style link sharing and commentary style application. But as I get deeper into the pure ID pieces, it is a deep well of work just getting that right - which we have a long way to go.   

I have looked at https://github.com/jonnycrunch/ipid and it looks like the way to get started.

I have just currently added both Github GIST proofs and Reddit post proofs and am just using a made up JSON format for the proof and signature. I look forward to learning more and implementing a proper DID and standard format.

**Comment by daviddahl** [comment]
Also, I did no tknow about `peer-star-identity`, thanks for the link.

**Comment by daviddahl** [comment]
Also, I did not know about `peer-star-identity`, thanks for the link.

**Comment by satazor** [comment]
> However, the eventual UX is key. Some folks don't believe the ID proofs concept is "easy enough" for the typical internet user.

I agree. The IDM is currently being designed by @andreforsousa and one of our main goals is to have a usable interface for the average internet user. In a few weeks we will have something that we can share. Do you have any mockups already designed?

I guess that @pgte already invited you to our WG bi-weekly meeting. We would love to get updates on this project and share knowledge/experiences there!

**Comment by daviddahl** [comment]
>Do you have any mockups already designed?

Not yet. I am have been focused on exposing the basic crypto primitives needed and supporting Github gists and Reddit posts as remote proof hosting. I would love to make it all seamless where the github API is leveraged to put the proof in a gist / reddit post, etc.

**Comment by w4ss4** [comment]
I would love to do some testing and finding bugs and i also got idea's we could add to it
